### PR TITLE
feat: Remove experimentalRunEvents flag

### PR DESCRIPTION
**Pull request by chrisbreiding**
- Addresses [TR-671](https://cypress-io.atlassian.net/browse/TR-671)

### User facing changelog

- The `experimentalRunEvents` configuration flag has been removed. It is no longer necessary to enable listening to run events in the plugins file.

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/3621
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [x] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by bahmutov** [comment]
Will these events be enabled in both interactive and non-interactive modes?

**Comment by chrisbreiding** [comment]
Just run mode for the time being.

**Comment by bahmutov** [comment]
I was hoping both, so I could switch code coverage plugin to it, otherwise no luck, has to happen in both modes https://github.com/cypress-io/code-coverage

**Comment by chrisbreiding** [comment]
That requires more discussion and work to implement. For now, we're just removing the flag.

File: packages/server/lib/config_options.ts [hunk 1]
```diff
@@ -74,11 +74,6 @@ export const options = [
     defaultValue: false,
     validation: v.isBoolean,
     isExperimental: true,
-  }, {
-    name: 'experimentalRunEvents',
-    defaultValue: false,
-    validation: v.isBoolean,
-    isExperimental: true,
```
Comment (flotwig) [inline packages/server/lib/config_options.ts:81]: would it be good to add this to `breakingOptions` at the bottom of this file, so users get a warning that it's no longer required to pass this option?

**Review by flotwig — APPROVED** [review]

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
I'm trying to see that the events work without the flag and I can't console.log any of the details. I know I have the changes because I see the new warning message. Am I missing something? Running via `yarn start`. 

```js
module.exports = (on, config) => {
  on('before:run', (details) => { 
    console.log(details) 
  })
  on('before:spec', (spec) => { 
    console.log(spec) 
  })
  on('after:spec', (spec, results) => { 
    console.log(spec, results) 
  })
  on('after:run', (results) => { 
    console.log(results) 
  })
}
```

Screen Shot 2021-03-01 at 7 59 48 PM

Also, when did this happen to our CSS? 😳 This error message on the plugins is completely weird.

**Comment by chrisbreiding** [comment]
> I'm trying to see that the events work without the flag and I can't console.log any of the details. I know I have the changes because I see the new warning message. Am I missing something? Running via yarn start.

These events only fire in `run` mode, so you won't see them with `yarn start`.

As for the CSS, I think the bootstrap upgrade would be the most likely culprit.

**Review by jennifer-shehane — APPROVED** [review]
@chrisbreiding We didn't upgrade bootstrap recently that I'm aware of. Do you have a PR link?

I'll open a new PR to fix the CSS issue.

**Comment by chrisbreiding** [comment]
My mistake, I thought we had merged the bootstrap upgrade, but we haven't. Maybe the node-sass change?

**Pull request merged**